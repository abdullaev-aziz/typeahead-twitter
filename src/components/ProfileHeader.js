import React from "react";
import "../styles/ProfileHeader.css";

function ProfileHeader({ user }) {
  const { name, screen_name, profile_image_url } = user;
  return (
    <div className="profile-header">
      <img alt="profile-image" src={profile_image_url} />
      <span className="name">{name}</span>
      <span>@{screen_name}</span>
    </div>
  );
}

export default ProfileHeader;
