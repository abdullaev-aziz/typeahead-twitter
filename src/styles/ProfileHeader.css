.profile-header {
  background: rgba(119, 174, 233, 0.364);
  display: flex;
  justify-content: flex-start;
  align-items: center;
  height: 50px;
  max-width: 600px;
  margin: auto;
  padding: 5px;
  border: 1px solid black;
}

.profile-header .name {
  font-weight: bold;
}

.profile-header span {
  display: inline-block;
  padding: 5px;
}
