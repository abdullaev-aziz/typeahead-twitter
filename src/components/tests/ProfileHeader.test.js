import React from 'react';
import { create } from 'react-test-renderer';

import ProfileHeader from '../ProfileHeader';

describe('ProfileHeader', ()=>{
    it("matches snapshot of ProfileHeader component", () => {
        const data = {
            name: "Sprout Social",
            screen_name: "SproutSocial",
            profile_image_url:
            "http://pbs.twimg.com/profile_images/625697856330952709/3dynAKiy_normal.png",
        }
        const element = create(
            
            <ProfileHeader user = {data}/>
        );
        expect(element.toJSON()).toMatchSnapshot();
      })
})