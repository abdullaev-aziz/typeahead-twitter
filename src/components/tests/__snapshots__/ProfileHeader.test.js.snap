// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHeader matches snapshot of ProfileHeader component 1`] = `
<div
  className="profile-header"
>
  <img
    alt="profile-image"
    src="http://pbs.twimg.com/profile_images/625697856330952709/3dynAKiy_normal.png"
  />
  <span
    className="name"
  >
    Sprout Social
  </span>
  <span>
    @
    SproutSocial
  </span>
</div>
`;
